apps/moose-console/src/components/query-interface.tsx

```diff
@@ -0,0 +1,284 @@
+}
+
+const insertSomeText = (insert: string, originalValue: string, ref: MutableRefObject<HTMLTextAreaElement>, setter: Dispatch<SetStateAction<string>>) => {
+    const selectionStart = ref.current.selectionStart;
+    const selectionEnd = ref.current.selectionEnd;
+ 
+    let newValue =
+      originalValue.substring(0, selectionStart) +
+      insert +
+      originalValue.substring(selectionEnd, originalValue.length);
+    setter(newValue);
+ };
+ 
+
+export default function QueryInterface({ table, related }: QueryInterfaceProps) {
+    // Create a ref to the textarea
+    const textareaRef = useRef<HTMLTextAreaElement>(null);
+    const tables = related.filter(t => !t.name.includes(".inner")).map(t => `${t.database}.${t.name}`)
+
+    const[value, setValue] = useState(`SELECT * FROM ${table.database}.${table.name} LIMIT 50;`)
+    const [results, setResults] = useState<any[]>()
+    const [sqlKeyWordCount, setSqlKeyWordCount] = useState(12)
+    const [tableCount, setTableCount] = useState(12)
+
+    useEffect(() => {
+        (
+            async () => {
+                const data = await runQuery(value)
+                setResults(data)
+            }
+        )()
+        
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (e.key === "Enter" && (e.ctrlKey || e.metaKey)) {
+                runQuery(value).then((results) => {
+                    e.preventDefault()
+                    setResults(results)
+                })
+            }
+        }
+
+        document.addEventListener("keydown", handleKeyDown)
+
+        return () => {
+            document.removeEventListener("keydown", handleKeyDown)
+        }
+    }, [])
+    
+
+    return (
+        <ResizablePanelGroup
+            className="h-full"
+            direction="vertical"
+        >
+            <ResizablePanel defaultSize={20}>
+            <ResizablePanelGroup direction="horizontal">
+                <ResizablePanel defaultSize={75}>
+                <div className="flex h-full">
+                    <Textarea ref={textareaRef} className="border-0 h-full font-mono" placeholder="type your query here" value={value} onChange={(e) => setValue(e.target.value)}/>
+                </div>
+                </ResizablePanel>
+                <ResizableHandle withHandle />
+                <ResizablePanel defaultSize={25}>
+                <div className="flex h-full px-4 w-full overflow-y-auto">
+                    <div className="grow">
+                        <div className="text-xs font-mono flex flex-row items-center">
+                            <span>SQL</span> <span className="flex-grow"/> 
+                            {
+                                sqlKeyWordCount < sqlKeyWords.length ? 
+                                <Button variant="ghost" onClick={() => setSqlKeyWordCount(sqlKeyWords.length)}>more</Button> : 
+                                <Button variant="ghost" onClick={() => setSqlKeyWordCount(5)}>less</Button>
```

Comment:
A bit weird that clicking "more" then "less" doesn't get me back to the same state.

Also I think I want to see tables before SQL keywords.

Reply:
Good catch